src/nccl_ofi_freelist.c

```diff
@@ -192,12 +192,14 @@ int nccl_ofi_freelist_add(nccl_ofi_freelist_t *freelist,
 	char *buffer;
 	struct nccl_ofi_freelist_block_t *block;
 
+	assert(freelist->max_entry_count == 0 ||
+		freelist->max_entry_count >= freelist->num_allocated_entries);
```

Comment:
I don't see the point of this assert.  The conditions are pretty well laid out below, and there are costs to extra asserts.

Reply:
I could foresee a situation in the future where elsewhere we update the logic of adding entries, or have a new function for adding entries, or somehow change `max_entry_count` on an existing freelist, and forget to update this logic. Without the assertion we'll just get a giant value for `allocation_count` and a difficult-to-find bug. So I like to be generous with asserts.

OK, if my little speech wasn't convincing, I'll remove the assertion and just leave the `<=` --> `==` change below. 😊

Reply:
you didn't convince me :)

Reply:
Removed!